server/models/Collective.js

```diff
@@ -858,6 +858,61 @@ Collective.prototype.getNextGoal = async function (until) {
   return nextGoal;
 };
 
+Collective.prototype.updateSocialLinks = async function (socialLinks) {
+  if (socialLinks.length > 10) {
+    throw new Error('account cannot set more than 10 social links');
+  }
+
+  if (socialLinks.length === 0) {
+    await models.SocialLink.destroy({
+      where: {
+        CollectiveId: this.id,
+      },
+    });
+    return [];
+  }
+
+  return await sequelize.transaction(async transaction => {
+    const existingLinks = await models.SocialLink.findAll({
+      where: {
+        CollectiveId: this.id,
+      },
+      transaction,
+      lock: true,
+    });
+
+    const removedLinks = differenceBy(existingLinks, socialLinks, sl => sl.url + sl.type);
```

Comment:
It's a minor issue, but this could conflict if there are two links like:
1. `https://url.com/GIT`
2. `https://url.com/`

I suggest using `differenceWith`:

```suggestion
    const isSameLink = (link1, link2) => link1.url === link2.url && link1.type === link2.type;
    const removedLinks = differenceWith(existingLinks, isSameLink);
```